src/OutputHash.js

```diff
@@ -128,23 +128,28 @@ OutputHash.prototype.apply = function apply(compiler) {
             const nonManifestChunks = this.chunks.filter(
                 chunk => !this.manifestFiles.includes(chunk.name)
             );
+
             const chunksByDependency = [];
 
             while (nonManifestChunks.length) {
                 let i = 0;
 
                 while (i < nonManifestChunks.length) {
-                    const current = nonManifestChunks[i];
-
-                    if (
-                        !current.parents
-                        || current.parents.length === 0
-                        || current.parents.every(p =>
-                            chunksByDependency.indexOf(p) !== -1
-                            || nonManifestChunks.indexOf(p) === -1
-                        )
-                    ) {
-                        chunksByDependency.push(current);
+                    const cur = nonManifestChunks[i];
+                    const parents = Array.from(cur.groupsIterable).reduce((acc, group) =>
+                        [...acc, ...group.getParents()], []
```

Comment:
/nit: We can use `acc.concat(group.getParents())` to avoid creating new arrays.

Reply:
Good point, will fix.